tools/hidl/hidlcomp.cpp

```diff
@@ -7034,38 +7122,89 @@ void HIDLcompiler::Process()
         write_esp_ng();
         write_esp_ng_cpp();
     }
-    
+
     if (isSCM)
         write_clarion_HRPC_interfaces();
 }
 
 
+void HIDLcompiler::processExecutionProfiling()
+{
+    //
+    // metrics execution profiling requires the memory header. Search the services and/or methods to determine
+    // if enabled in any. If so, write the include statement
+    EspServInfo *si;
+    bool profilingEnabled = false;
+    StrBuffer tmp;
+    for (si=servs; si; si=si->next)
+    {
+        StrBuffer serviceProfilingOptions;
+        bool serviceProfileExecutionEnabled = si->getMetaStringValue(serviceProfilingOptions,"profile_execution");
+
+        //
+        // At the HIDL top level, execution profiling is enable if any service is enabled
+        executionProfilingEnabled |= serviceProfileExecutionEnabled;
+
+        //
+        // Go through each method and save any profile information to make if faster later when
+        // generating code. Note that the default for the method is either what is specified in the
+        // method or what is set at the service level. Still track the executionProfileEnabled flag for HIDL at the top
+        EspMethodInfo *mthi=NULL;
+        for (mthi=si->methods;mthi!=NULL;mthi=mthi->next)
+        {
+            //
+            // Collect method profile execution values
+            StrBuffer methodProfilingOptions;
+            bool methodProfileExecutionEnabled = mthi->getMetaStringValue(methodProfilingOptions, "profile_execution");
+            executionProfilingEnabled |= methodProfileExecutionEnabled;   // again, if a method is enabled, set top flag
+            if (serviceProfileExecutionEnabled || methodProfileExecutionEnabled)
+            {
+                if (!mthi->getMetaInt("no_profile_execution"))
+                {
+                    mthi->setEnableExecutionProfiling(true);
+                    if (!mthi->setExecutionProfilingOptions(methodProfileExecutionEnabled ? methodProfilingOptions.str() : serviceProfilingOptions.str()))
+                    {
+                        throw "Invalid execution profile options detected";
```

Comment:
I followed the standard used elsewhere in the code. If you want, I can build a string with the name of the method and throw that. You are right that it may leak, but it's probably not an issue since this is a tool that only runs at build time. Please let me know if you want that change before final approval.